packages/wallet-sdk/src/connection/WalletLinkWebSocket.ts

```diff
@@ -94,7 +94,11 @@ export class WalletLinkWebSocket {
       return;
     }
     this.clearWebSocket();
+
     this.connectionStateListener?.(ConnectionState.DISCONNECTED);
+    this.connectionStateListener = undefined;
+    this.incomingDataListener = undefined;
```

Comment:
clean up after calling `this.connectionStateListener?.(ConnectionState.DISCONNECTED);`